pkg/agent/flowexporter/connections/conntrack.go

```diff
@@ -74,6 +105,14 @@ func filterAntreaConns(conns []*flowexporter.Connection, nodeConfig *config.Node
 				continue
 			}
 		}
+
+		if protocolFilterSpecified {
+			protocol := string(serviceProtocolMap[conn.FlowKey.Protocol])
+			if !slices.Contains(protocolFilter, protocol) {
+				continue
+			}
```

Comment:
actually it may be better to convert the protocol list to integers, to avoid the map lookup + slice search. This way we would just have a slice search.